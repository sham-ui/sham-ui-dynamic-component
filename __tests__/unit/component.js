import renderer, { compile } from 'sham-ui-test-helpers';
import DynamicComponent from '../../src/component';

it( 'empty options', () => {
    const meta = renderer( DynamicComponent, {} );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'component', () => {
    const meta = renderer( DynamicComponent, {
        component: compile`<div>Foo component content</div>`
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'component options', () => {
    const meta = renderer( DynamicComponent, {
        component: compile`<div>Foo component content. {{text}}</div>`,
        text: 'Extra text'
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
    meta.component.update( {
        text: 'Another extra text'
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'update component', () => {
    const meta = renderer( DynamicComponent, {
        component: compile`<div>Foo component content. {{text}}</div>`,
        text: 'Extra text'
    } );
    expect( meta.toJSON() ).toMatchSnapshot();

    meta.component.update( {
        component: compile`<div>Bar component content. {{text}}</div>`
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

describe( 'blocks', () => {
    beforeEach( () => window.DynamicComponent = DynamicComponent );
    afterEach( () => delete window.DynamicComponent );

    it( 'blocks', () => {
        const meta = renderer(
            compile`
            <DynamicComponent 
                component={{component}}  
                text={{text}}
            >
                {% block 'title' %}
                    Title for component
                {% endblock %}
            </DynamicComponent>
        `,
            {
                component: compile`
                    <h3>{% defblock 'title' %}</h3>
                    <div>Foo component content. {{text}}</div>
                `,
                text: 'Extra text'
            }
        );
        expect( meta.toJSON() ).toMatchSnapshot();

        meta.component.update( {
            component: compile`<div>Component without block</div>`
        } );
        expect( meta.toJSON() ).toMatchSnapshot();
    } );
} );

