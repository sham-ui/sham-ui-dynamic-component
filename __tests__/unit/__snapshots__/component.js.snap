// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocks 1`] = `
Object {
  "Constructor": "dummy",
  "Options": Object {
    "ID": "component",
    "component": [Function],
    "text": "Extra text",
  },
  "html": "
<h3> Title for component
  <!--title-->
</h3>
<div>Foo component content. Extra text</div>
<!--DynamicComponent-->
",
}
`;

exports[`blocks 2`] = `
Object {
  "Constructor": "dummy",
  "Options": Object {
    "ID": "component",
    "component": [Function],
    "text": "Extra text",
  },
  "html": "
<div>Component without block</div>
<!--DynamicComponent-->
",
}
`;

exports[`component 1`] = `
Object {
  "Constructor": "Dynamic",
  "Options": Object {
    "ID": "component",
    "component": [Function],
  },
  "html": "<div>Foo component content</div>",
}
`;

exports[`component options 1`] = `
Object {
  "Constructor": "Dynamic",
  "Options": Object {
    "ID": "component",
    "component": [Function],
    "text": "Extra text",
  },
  "html": "<div>Foo component content. Extra text</div>",
}
`;

exports[`component options 2`] = `
Object {
  "Constructor": "Dynamic",
  "Options": Object {
    "ID": "component",
    "component": [Function],
    "text": "Another extra text",
  },
  "html": "<div>Foo component content. Another extra text</div>",
}
`;

exports[`empty options 1`] = `
Object {
  "Constructor": "Dynamic",
  "Options": Object {
    "ID": "component",
    "component": undefined,
  },
  "html": "",
}
`;

exports[`update component 1`] = `
Object {
  "Constructor": "Dynamic",
  "Options": Object {
    "ID": "component",
    "component": [Function],
    "text": "Extra text",
  },
  "html": "<div>Foo component content. Extra text</div>",
}
`;

exports[`update component 2`] = `
Object {
  "Constructor": "Dynamic",
  "Options": Object {
    "ID": "component",
    "component": [Function],
    "text": "Extra text",
  },
  "html": "<div>Bar component content. Extra text</div>",
}
`;
